src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmParticipantSpec.scala

```diff
@@ -0,0 +1,353 @@
+        gridAgentWithParticipants,
+        firstSweepNo
+      )
+
+      // the gridAgentWithParticipants has received an AssetPowerChangedMessage
+      // before requesting power from the superiorGrid
+      superiorGridAgent.expectGridPowerProvision(
+        Seq(
+          ExchangePower(
+            supNodeA.getUuid,
+            Quantities.getQuantity(135.90837346741768, MEGAWATT),
+            Quantities.getQuantity(60.98643348675892, MEGAVAR)
+          )
+        )
+      )
+
+      // before the second sweep the gridAgentWithParticipants will receive an AssetPowerUnchangedMessage
+      // we start a second sweep by asking for next sweep values which should trigger the whole procedure again
+      val secondSweepNo = firstSweepNo + 1
+
+      superiorGridAgent.requestGridPower(
+        gridAgentWithParticipants,
+        secondSweepNo
+      )
+
+      // the agent now should ask for updated slack voltages from the superior grid
+      val secondSlackAskSender =
+        superiorGridAgent.expectSlackVoltageRequest(secondSweepNo)
+
+      // the superior grid would answer with updated slack voltage values
+      superiorGridAgent.gaProbe.send(
+        secondSlackAskSender,
+        ProvideSlackVoltageMessage(
+          secondSweepNo,
+          Seq(
+            ExchangeVoltage( // this one should currently be ignored anyways
+              supNodeA.getUuid,
+              Quantities.getQuantity(380, KILOVOLT),
+              Quantities.getQuantity(0, KILOVOLT)
+            )
+          )
+        )
+      )
+
+      // here the gridAgentWithParticipants has received a second AssetPowerUnchangedMessage
+      // we expect that the GridAgent unstashes the messages and return a value for our power request
+      superiorGridAgent.expectGridPowerProvision(
+        Seq(
+          ExchangePower(
+            supNodeA.getUuid,
+            Quantities.getQuantity(135.90837346741768, MEGAWATT),
+            Quantities.getQuantity(60.98643348675892, MEGAVAR)
+          )
+        )
+      )
+
+      // normally the superior grid agent would check whether the power flow calculation converges and would
+      // send a CompletionMessage to the scheduler and a FinishGridSimulationTrigger to the inferior grid agent
+      // after the convergence
+      // (here we do it by hand)
+      superiorGridAgent.gaProbe.send(
+        scheduler.ref,
+        CompletionMessage(
+          5,
+          Some(
+            Seq(
+              ScheduleTriggerMessage(
+                ActivityStartTrigger(7200L),
+                superiorGridAgent.gaProbe.ref
+              )
+            )
+          )
+        )
+      )
+
+      scheduler.expectMsg(
+        CompletionMessage(
+          5,
+          Some(
+            Seq(
+              ScheduleTriggerMessage(
+                ActivityStartTrigger(7200L),
+                superiorGridAgent.gaProbe.ref
+              )
+            )
+          )
+        )
+      )
```

Comment:
Ok this is something that I overlooked with the first review, but this message exchange between test probes is pointless (same as in `DBFSAlgorithmCenGridSpec`)